ui: A collection of fixes the network debugging page

BramGruneir opened the pull request:
- Fix the case in which longs had no value during calculation causing the page
to not load.
- Fix latency page to only display when there is at least 2 nodes worth of
latencies to display.
- Add a loading spinner to the network diagnostics page.
- Fix the loading spinner to accept an array of children instead of just a
single child element.
- Add a new top padded style so the loading spinner is not so cramped on the
page.

Fixes #23552.

Release note (admin ui change): The network diagnositcs report will no longer
crash when the latencies are very small or on a single node cluster.

BramGruneir committed:
ui: A collection of fixes for the network debugging page

- Fix the case in which longs had no value during calculation causing the page
to not load.
- Fix latency page to only display when there is at least 2 nodes worth of
latencies to display.
- Add a loading spinner to the network diagnostics page.
- Fix the loading spinner to accept an array of children instead of just a
single child element.
- Add a new top padded style so the loading spinner is not so cramped on the
page.
- Fix a crash when a new node doesn't yet have any activity.
- Replace all not yet calculated latencies with a `?`.
- Don't display a legend if there is only one value (and hence no stddev).

Fixes #23552.

Release note (admin ui change): The network diagnositcs report will no longer
crash when the latencies are very small or on a single node cluster.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/23844)
<!-- Reviewable:end -->

couchand commented on pkg/ui/src/views/reports/containers/network/index.tsx:
```diff
@@ -399,20 +405,37 @@ class Network extends React.Component<NetworkProps, {}> {
-      <div className="section">
-        <h1>Network Diagnostics</h1>
-        <NodeFilterList nodeIDs={filters.nodeIDs} localityRegex={filters.localityRegex} />
+      <div>
```
this wrapper div is unfortunate.  we should try to upgrade to React 16 soon after 2.0, so we can use this: https://reactjs.org/blog/2017/11/28/react-v16.2.0-fragment-support.html

In this case I think returning a list would probably be ok...

couchand commented on pkg/ui/src/views/shared/components/loading/index.tsx:
```diff
@@ -18,5 +18,9 @@ export default function Loading(props: LoadingProps) {
     return <div className={props.className} style={image} />;
   }
-  return props.children as JSX.Element;
+  return (
```
It might be worth commenting on why this wrapper div is necessary so we don't accidentally remove it.  That comment might also mention that after upgrading to React 16 we could convert it to a fragment.

couchand reviewed (commented):
While verifying this one time I got a console error of `activity is undefined`, but I can't repro that now.  Not sure if that rings a bell...

BramGruneir commented:
PTAL.  I've added a few new fixes.  `activity is undefined` should be fixed.

Specifically, I've added:
- Fix a crash when a new node doesn't yet have any activity.
- Replace all not yet calculated latencies with a `?`.
- Don't display a legend if there is only one value (and hence no stddev).

As well as now there are `?` when we don't have the latency instead of showing a 0.

![Screen Shot 2018-03-14 at 15.20.53.png](https://files.reviewableusercontent.io/images/reviewable/1614265/cuTBwXxORj_vImK7TAreFo3R/Screen%20Shot%202018-03-14%20at%2015.20.53.png)

---

Review status: 0 of 4 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/23844#-:-L7_yQMsacpFLNtdUn9T:b7e3y5x)*
<!-- Sent from Reviewable.io -->

couchand commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> thanks!

---

Reviewed 2 of 4 files at r1, 2 of 2 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/23844#-:-L7aH_lWme9yiheyFu-w:brfqlux)*
<!-- Sent from Reviewable.io -->
